src/main.ts

```diff
@@ -73,18 +74,38 @@ class LocalTableland {
     // TODO: I don't think this is doing anything anymore...
     this.#_cleanup();
 
+    // check if the hardhat port is in use and retry 5 times (1 second between each)
+    const hardhatPort = 8545;
+    let retries = 0;
+    while (retries < 5) {
+      const portInUse = await checkPortInUse(hardhatPort);
+      if (!portInUse) break;
+
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+      retries++;
+    }
+    // throw if all retries failed and port is in use
+    if (retries === 5) {
+      throw new Error(
+        `cannot start a local chain due to port ${hardhatPort} in use`
+      );
+    }
+
+    // You *must* store these in `process.env` to access within the hardhat subprocess
+    process.env.HARDHAT_NETWORK = "hardhat";
+    process.env.HARDHAT_UNLIMITED_CONTRACT_SIZE = "true";
+    process.env.HARDHAT_PORT = hardhatPort.toString();
+
     // Run a local hardhat node
     this.registry = spawn(
       isWindows() ? "npx.cmd" : "npx",
-      ["hardhat", "node"],
+      ["hardhat", "node", "--port", hardhatPort.toString()], // TODO: use a fallback port on conflicts vs. failing on retries
```

Comment:
this doesn't change any behavior but at least provides the option to use custom port fallbacks, if desired in the future.